tests/lang/s02_core/s99_coroutines/t11_suspend_coroutine.nim

```diff
@@ -0,0 +1,43 @@
+discard """
+  output: '''
+outside: 1
+coro: 1
+outside: 2
+coro: 2
+outside: 3
+coro: 3'''
+"""
+
+## A coroutine can be suspended without returning or raising an exception.
+## This is achieved by calling the built-in ``suspend`` routine, which expects
+## a ``CoroutineBase`` as the parameter -- the instance passed to ``suspend``
+## is what ``resume`` will return. It is legal to pass the currently running
+## coroutine (i.e., `self`) to ``suspend``.
+
+# IDEA: instead of the ``suspend`` routine, the ``yield`` keyword could be
+#       re-used, which could be much more convenient to use. It'd also be
+#       a bit easier to implement
+
+proc coro() {.coroutine.} =
+  echo "coro: 1"
+  suspend(self)
```

Comment:
looks like i can call suspend on any coroutine at any place but this is wrong

``` nim
proc a() {.coroutine.} =
  var count = 0
  while true:
    count += 1
let instance = a()
var t: Thread[Coroutine]
createThread(t, resume, instance)
suspend(instance)
```

Reply:
Yep, there was no check in place to prevent usage of `suspend` outside of coroutines. I've added both a specification test clarifying the behaviour, as well as the missing check (the error message is not yet proper).